cartographer_ros/cartographer_ros/node.cc

```diff
@@ -486,14 +486,17 @@ bool Node::HandleFinishTrajectory(
     ::cartographer_ros_msgs::FinishTrajectory::Request& request,
     ::cartographer_ros_msgs::FinishTrajectory::Response& response) {
   carto::common::MutexLocker lock(&mutex_);
-  return FinishTrajectoryUnderLock(request.trajectory_id);
+  bool success = FinishTrajectoryUnderLock(request.trajectory_id);
+  response.success = success;
+  return success;
```

Comment:
This is well odd to me. Previously, we returned it directly. IIRC, when you return false here, the caller simply gets a failed service call, so you can detect it that way. I don't see the point of the `success` field, if you're still returning false when it fails. `success` is always going to be true, or else you get a failed call, IIRC again.

Reply:
(Ok, sorry for the confusion, this is going a bit off-topic from the initial motivation 😃 )

You're right, doesn't change the situation for the service caller so far. However, a `false` return value, followed by a failed service call, does not indicate for the caller whether it was a logical error (here e.g. wrong trajectory ID) or if sth in the communication failed. (see also [here](https://answers.ros.org/question/268051/why-isnt-the-response-filled-if-my-service-returns-false/?answer=268100#post-id-268100))

So my opinion now is: it looks cleaner to return always `true` to indicate a service was properly *executed*, but at the same to have additional response fields that indicate for the caller whether the service *did the right thing*. This is also true for `StartTrajectory`. The failed service calls induced by `false` return values have zero information for a caller that can't access the logs directly.

You're right about `WriteState`, I'd only like to add the new log message here.

Reply:
You're right in the sense that it would make differentiating between network problems and bad arguments to `FinishTrajectory` easier.